Added Makefile to generate png images

redvimo opened the pull request:
Hi, Lazka.
Makefile all done. Im still learning how to use make, so I made it simple for me to understand. 
I was not sure whether to execute the Makefile before doing the commit, so i just did it anyway so you can see that it worked. 
Thanks.

lazka commented on quodlibet/quodlibet/images/hicolor/Makefile:
```diff
@@ -0,0 +1,82 @@
+QL_SOURCE=scalable/apps/quodlibet.svg
+QL_DEST=quodlibet.png
+
+QL_SYM_SOURCE=scalable/apps/quodlibet-symbolic.svg
```
The "-symbolic" icons can't be converted to png. They are a special svg variant that allows color rewriting.

redvimo replied:
So here I should remove the targets and variables for the "-symbolic" icons? Meaning don't convert them to png? Why do you say can't be converted? Inkscape seems to have converted it to png. Do you mean they should not be converted?

lazka replied:
They can be converted, but gtk+ is not able to use them. Just remove the targets/variables.

redvimo replied:
Ok cool

lazka commented on quodlibet/quodlibet/images/hicolor/Makefile:
```diff
@@ -0,0 +1,82 @@
+APPS_DIR=16x16/apps 24x24/apps 32x32/apps 48x48/apps 64x64/apps
+STATUS_DIR=16x16/status 24x24/status 32x32/status 48x48/status 64x64/status
+
+all: mkdestdir png16 png24 png32 png48 png64
```
The dependencies might not get executed in the given order, you have to move "mkdestdir" to the pngXX targets.

redvimo replied:
If I understand correctly, you mean do this: 
...
png16: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 16x16/apps 16x16/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 16x16/apps/$(QL_DEST) -w 16 -h 16 ; \
	inkscape $(EF_SOURCE) --export-png 16x16/apps/$(EF_DEST) -w 16 -h 16 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 16x16/status/$(QL_MISSING_COVER_DEST) \
		-w 16 -h 16 ; \
	};

png24: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 24x24/apps 24x24/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 24x24/apps/$(QL_DEST) -w 24 -h 24 ; \
	inkscape $(EF_SOURCE) --export-png 24x24/apps/$(EF_DEST) -w 24 -h 24 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 24x24/status/$(QL_MISSING_COVER_DEST) \
		-w 24 -h 24 ; \
	};
...

redvimo replied:
The formatting went off, because copy and pasted it. I tried it with the "&&" with a "{list;}". Seems to work, but is that the right way to have the png files commands executed only if the mkdir command is successful. Or does it not matter?

lazka replied:
looks good. make should fail automatically if one command fails. No need for && I think

redvimo replied:
Thanks

redvimo commented:
Ok check this new makefile out.. It works but executes all targets even if they exist. Still trying to figure it out. Maybe I should change the pngXX targets to the actual filepath?

`
QL_SOURCE=scalable/apps/quodlibet.svg
QL_DEST=quodlibet.png

QL_MISSING_COVER_SOURCE=scalable/status/quodlibet-missing-cover.svg
QL_MISSING_COVER_DEST=quodlibet-missing-cover.png

EF_SOURCE=scalable/apps/exfalso.svg
EF_DEST=exfalso.png

APPS_DIR=16x16/apps 24x24/apps 32x32/apps 48x48/apps 64x64/apps
STATUS_DIR=16x16/status 24x24/status 32x32/status 48x48/status 64x64/status

all: png16 png24 png32 png48 png64

png16: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 16x16/apps 16x16/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 16x16/apps/$(QL_DEST) -w 16 -h 16 ; \
	inkscape $(EF_SOURCE) --export-png 16x16/apps/$(EF_DEST) -w 16 -h 16 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 16x16/status/$(QL_MISSING_COVER_DEST) \
		-w 16 -h 16 ; \
	};

png24: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 24x24/apps 24x24/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 24x24/apps/$(QL_DEST) -w 24 -h 24 ; \
	inkscape $(EF_SOURCE) --export-png 24x24/apps/$(EF_DEST) -w 24 -h 24 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 24x24/status/$(QL_MISSING_COVER_DEST) \
		-w 24 -h 24 ; \
	};

png32: $(QL_SOURCE) $(EL_SOURCE)
	mkdir -p 32x32/apps 32x32/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 32x32/apps/$(QL_DEST) -w 32 -h 32 ; \
	inkscape $(EF_SOURCE) --export-png 32x32/apps/$(EF_DEST) -w 32 -h 32 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 32x32/status/$(QL_MISSING_COVER_DEST) \
		-w 32 -h 32 ; \
	};

png48: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 48x48/apps 48x48/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 48x48/apps/$(QL_DEST) -w 48 -h 48 ; \
	inkscape $(EF_SOURCE) --export-png 48x48/apps/$(EF_DEST) -w 48 -h 48 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 48x48/status/$(QL_MISSING_COVER_DEST) \
		-w 48 -h 48 ; \
	};

png64: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 64x64/apps 64x64/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 64x64/apps/$(QL_DEST) -w 64 -h 64 ; \
	inkscape $(EF_SOURCE) --export-png 64x64/apps/$(EF_DEST) -w 64 -h 64 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 64x64/status/$(QL_MISSING_COVER_DEST) \
		-w 64 -h 64 ; \
	};

.PHONY: clean

clean:
	rm -R 16x16 24x24 32x32 48x48 64x64

`

redvimo commented:
```
QL_SOURCE=scalable/apps/quodlibet.svg
QL_DEST=quodlibet.png

QL_MISSING_COVER_SOURCE=scalable/status/quodlibet-missing-cover.svg
QL_MISSING_COVER_DEST=quodlibet-missing-cover.png

EF_SOURCE=scalable/apps/exfalso.svg
EF_DEST=exfalso.png

APPS_DIR=16x16/apps 24x24/apps 32x32/apps 48x48/apps 64x64/apps
STATUS_DIR=16x16/status 24x24/status 32x32/status 48x48/status 64x64/status

all: png16 png24 png32 png48 png64

png16: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 16x16/apps 16x16/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 16x16/apps/$(QL_DEST) -w 16 -h 16 ; \
	inkscape $(EF_SOURCE) --export-png 16x16/apps/$(EF_DEST) -w 16 -h 16 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 16x16/status/$(QL_MISSING_COVER_DEST) \
		-w 16 -h 16 ; \
	};

png24: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 24x24/apps 24x24/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 24x24/apps/$(QL_DEST) -w 24 -h 24 ; \
	inkscape $(EF_SOURCE) --export-png 24x24/apps/$(EF_DEST) -w 24 -h 24 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 24x24/status/$(QL_MISSING_COVER_DEST) \
		-w 24 -h 24 ; \
	};

png32: $(QL_SOURCE) $(EL_SOURCE)
	mkdir -p 32x32/apps 32x32/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 32x32/apps/$(QL_DEST) -w 32 -h 32 ; \
	inkscape $(EF_SOURCE) --export-png 32x32/apps/$(EF_DEST) -w 32 -h 32 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 32x32/status/$(QL_MISSING_COVER_DEST) \
		-w 32 -h 32 ; \
	};

png48: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 48x48/apps 48x48/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 48x48/apps/$(QL_DEST) -w 48 -h 48 ; \
	inkscape $(EF_SOURCE) --export-png 48x48/apps/$(EF_DEST) -w 48 -h 48 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 48x48/status/$(QL_MISSING_COVER_DEST) \
		-w 48 -h 48 ; \
	};

png64: $(QL_SOURCE) $(EF_SOURCE)
	mkdir -p 64x64/apps 64x64/status && \
	{ \
	inkscape $(QL_SOURCE) --export-png 64x64/apps/$(QL_DEST) -w 64 -h 64 ; \
	inkscape $(EF_SOURCE) --export-png 64x64/apps/$(EF_DEST) -w 64 -h 64 ; \
	inkscape $(QL_MISSING_COVER_SOURCE) \
		--export-png 64x64/status/$(QL_MISSING_COVER_DEST) \
		-w 64 -h 64 ; \
	};

.PHONY: clean

clean:
	rm -R 16x16 24x24 32x32 48x48 64x64

```

lazka commented:
APPS_DIR, STATUS_DIR is unused now

redvimo commented:
@lazka Ok will remove those variables. Should i remove the "&& {list;}" parts? I only used it so that the command list gets executed as a whole if the mkdir command doesnt fail. 
Will find out more on that..

redvimo commented:
After making the changes must i do a new pull request, and then you will close this one?

redvimo commented:
Greetings...
I made some changes to the last Makefile, including your suggestions. Let me know if this is better.

1. Separated commands into it's own target, because if I delete one of the generated png files then another run of "make all" will generate that whole group of png files, not just the one I deleted.

2. Changed png targets to be files, so that make does not generate target png files unnecessarily.

```
QL_SOURCE=scalable/apps/quodlibet.svg
QL_DEST=quodlibet.png

EF_SOURCE=scalable/apps/exfalso.svg
EF_DEST=exfalso.png

QL_MC_SOURCE=scalable/status/quodlibet-missing-cover.svg
QL_MC_DEST=quodlibet-missing-cover.png

PNG16_APPS_DIR=16x16/apps
PNG16_STATUS_DIR=16x16/status
PNG16_QL_DEST=$(PNG16_APPS_DIR)/$(QL_DEST)
PNG16_EF_DEST=$(PNG16_APPS_DIR)/$(EF_DEST)
PNG16_MC_DEST=$(PNG16_STATUS_DIR)/$(QL_MC_DEST)

PNG24_APPS_DIR=24x24/apps
PNG24_STATUS_DIR=24x24/status
PNG24_QL_DEST=$(PNG24_APPS_DIR)/$(QL_DEST)
PNG24_EF_DEST=$(PNG24_APPS_DIR)/$(EF_DEST)
PNG24_MC_DEST=$(PNG24_STATUS_DIR)/$(QL_MC_DEST)

PNG32_APPS_DIR=32x32/apps
PNG32_STATUS_DIR=32x32/status
PNG32_QL_DEST=$(PNG32_APPS_DIR)/$(QL_DEST)
PNG32_EF_DEST=$(PNG32_APPS_DIR)/$(EF_DEST)
PNG32_MC_DEST=$(PNG32_STATUS_DIR)/$(QL_MC_DEST)

PNG48_APPS_DIR=48x48/apps
PNG48_STATUS_DIR=48x48/status
PNG48_QL_DEST=$(PNG48_APPS_DIR)/$(QL_DEST)
PNG48_EF_DEST=$(PNG48_APPS_DIR)/$(EF_DEST)
PNG48_MC_DEST=$(PNG48_STATUS_DIR)/$(QL_MC_DEST)

PNG64_APPS_DIR=64x64/apps
PNG64_STATUS_DIR=64x64/status
PNG64_QL_DEST=$(PNG64_APPS_DIR)/$(QL_DEST)
PNG64_EF_DEST=$(PNG64_APPS_DIR)/$(EF_DEST)
PNG64_MC_DEST=$(PNG64_STATUS_DIR)/$(QL_MC_DEST)


all: png16 png24 png32 png48 png64

png16: $(PNG16_QL_DEST) $(PNG16_EF_DEST) $(PNG16_MC_DEST)

png24: $(PNG24_QL_DEST) $(PNG24_EF_DEST) $(PNG24_MC_DEST)

png32: $(PNG32_QL_DEST) $(PNG32_EF_DEST) $(PNG32_MC_DEST)

png48: $(PNG48_QL_DEST) $(PNG48_EF_DEST) $(PNG48_MC_DEST)

png64: $(PNG64_QL_DEST) $(PNG64_EF_DEST) $(PNG64_MC_DEST)

$(PNG16_QL_DEST): $(QL_SOURCE)
	mkdir -p $(PNG16_APPS_DIR) && \
	inkscape $(QL_SOURCE) --export-png $(PNG16_QL_DEST) -w 16 -h 16 ;

$(PNG16_EF_DEST): $(EF_SOURCE)
	mkdir -p $(PNG16_APPS_DIR) && \
	inkscape $(EF_SOURCE) --export-png $(PNG16_EF_DEST) -w 16 -h 16 ;

$(PNG16_MC_DEST): $(MC_SOURCE)
	mkdir -p $(PNG16_STATUS_DIR) && \
	inkscape $(QL_MC_SOURCE) --export-png $(PNG16_MC_DEST) -w 16 -h 16 ;

$(PNG24_QL_DEST): $(QL_SOURCE)
	mkdir -p $(PNG24_APPS_DIR) && \
	inkscape $(QL_SOURCE) --export-png $(PNG24_QL_DEST) -w 24 -h 24 ;

$(PNG24_EF_DEST): $(EF_SOURCE)
	mkdir -p $(PNG24_APPS_DIR) && \
	inkscape $(EF_SOURCE) --export-png $(PNG24_EF_DEST) -w 24 -h 24 ;

$(PNG24_MC_DEST): $(MC_SOURCE)
	mkdir -p $(PNG24_STATUS_DIR) && \
	inkscape $(QL_MC_SOURCE) --export-png $(PNG24_MC_DEST) -w 24 -h 24 ;

$(PNG32_QL_DEST): $(QL_SOURCE)
	mkdir -p $(PNG32_APPS_DIR) && \
	inkscape $(QL_SOURCE) --export-png $(PNG32_QL_DEST) -w 32 -h 32 ;

$(PNG32_EF_DEST): $(EF_SOURCE)
	mkdir -p $(PNG32_APPS_DIR) && \
	inkscape $(EF_SOURCE) --export-png $(PNG32_EF_DEST) -w 32 -h 32 ;

$(PNG32_MC_DEST): $(MC_SOURCE)
	mkdir -p $(PNG32_STATUS_DIR) && \
	inkscape $(QL_MC_SOURCE) --export-png $(PNG32_MC_DEST) -w 32 -h 32 ;

$(PNG48_QL_DEST): $(QL_SOURCE)
	mkdir -p $(PNG48_APPS_DIR) && \
	inkscape $(QL_SOURCE) --export-png $(PNG48_QL_DEST) -w 48 -h 48 ;

$(PNG48_EF_DEST): $(EF_SOURCE)
	mkdir -p $(PNG48_APPS_DIR) && \
	inkscape $(EF_SOURCE) --export-png $(PNG48_EF_DEST) -w 48 -h 48 ;

$(PNG48_MC_DEST): $(MC_SOURCE)
	mkdir -p $(PNG48_STATUS_DIR) && \
	inkscape $(QL_MC_SOURCE) --export-png $(PNG48_MC_DEST) -w 48 -h 48 ;

$(PNG64_QL_DEST): $(QL_SOURCE)
	mkdir -p $(PNG64_APPS_DIR) && \
	inkscape $(QL_SOURCE) --export-png $(PNG64_QL_DEST) -w 64 -h 64 ;

$(PNG64_EF_DEST): $(EF_SOURCE)
	mkdir -p $(PNG64_APPS_DIR) && \
	inkscape $(EF_SOURCE) --export-png $(PNG64_EF_DEST) -w 64 -h 64 ;

$(PNG64_MC_DEST): $(MC_SOURCE)
	mkdir -p $(PNG64_STATUS_DIR) && \
	inkscape $(QL_MC_SOURCE) --export-png $(PNG64_MC_DEST) -w 64 -h 64 ;

.PHONY: clean

clean:
	rm -f -R ./16x16

```

lazka commented:
> After making the changes must i do a new pull request, and then you will close this one?

You can just continue pushing to this one.

redvimo commented:
Hey, i completed the makefile. This is what I did using git locally, please tell me if the steps are correct.
1. Cloned your master branch
2. Changed the makefile
3. Synchronized my local copy with your upstream, using the commands:
```
git fetch upstream
git checkout master
git merge upstream/master
```
4. Commit the changes using: `git commit -a`
5. Pushed the changes to my forks master branch using: `git push origin master`

lazka commented:
Thanks

redvimo commented:
Great... Thank you
